waiter/src/waiter/websocket.clj

```diff
@@ -72,8 +72,8 @@
 (defn request-handler
   "Handler for websocket requests.
    It populates the kerberos credentials and invokes process-request-fn."
-  [password process-request-fn {:keys [headers] :as request}]
-  (let [auth-cookie (-> headers (get "cookie") str auth/get-auth-cookie-value) ;; auth-cookie is assumed to be valid
+  [password process-request-fn {:keys [cookies headers] :as request}]
+  (let [auth-cookie (get-in cookies [auth/AUTH-COOKIE-NAME :value]) ;; auth-cookie is assumed to be valid
```

Comment:
I'd prefer to have a `(auth/get-auth-cookie cookies)` function? This snippet was repeated in `spnego.clj`